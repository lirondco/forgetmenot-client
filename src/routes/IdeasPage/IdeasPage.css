.list_name {
    font-family: 'Mina', sans-serif;
    padding: 5%;
}

.empty_list {
    padding-bottom: 100%;
    color: red;
}