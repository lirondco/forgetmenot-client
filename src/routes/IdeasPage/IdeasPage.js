import React, { Component } from 'react'
import IdeasContext from '../../contexts/IdeasContext'
import IdeasApiService from '../../services/ideas-api-service'
import { Section } from '../../components/Utils/Utils'
import Accordion from '../../components/Accordion/Accordion'
import { Link } from 'react-router-dom'
import './IdeasPage.css'

export default class IdeasPage extends Component {
  static defaultProps = {
    match: { params: {} },
    history: {
        push: () => {},
      },
  }

  static contextType = IdeasContext

  componentDidMount() {
//retrieve data from the server upon the component's mounting
    const { listId } = this.props.match.params
    IdeasApiService.getList(listId)
      .then(this.context.setList)
      .then(this.context.clearError)
      .catch(this.context.setError)
    IdeasApiService.getListIdeas(listId)
      .then(this.context.setIdeas)
      .then(this.context.clearError)
      .catch(this.context.setError)
  }

  componentWillUnmount() {
//clears the context out when done
    this.context.clearList()
  }

  handleDeleteSuccess = idea => {
    const { history } = this.props
    history.push(`/lists/${this.context.list.id}`)
    this.context.deleteIdea(idea)
  }

  renderList() {
    const { list, ideas } = this.context
    return <>
      <h2 className='list_name'>{list.name}</h2>
      <Ideas ideas={ideas} deleteFunction={this.handleDeleteSuccess} />
    </>
  }

  render() {
    const { error, list={} } = this.context
    let content
    if (error) {
//adds a layer for error handling
      content = (error.error === `List doesn't exist`)
        ? <p className='red'>List not found</p>
        : <p className='red'>There was an error</p>
    } else if (!list.id) {
      content = <div className='loading' />
    } else {
      content = this.renderList()
    }
    return (
      <Section className='IdeasPage'>
        {content}
      </Section>
    )
  }
}

function Ideas({ ideas = [], deleteFunction }) {
// adds a message with a link to the home page when a list is empty
    if (ideas.length === 0) {
        return (
            <div className='empty_list'>
            <p>This list is empty!</p>
            <Link to='/'>Go back to home page</Link>
            </div>
        )
    }
  return (
      ideas.map(idea =>
          <Accordion
           onDeleteSuccess={() => deleteFunction(idea.id)}
           key={idea.id}
           id={idea.id}
           title={idea.name}
           content={idea.content} />
      )
  )
}
