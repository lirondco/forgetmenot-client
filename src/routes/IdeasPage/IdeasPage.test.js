import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import IdeasPage from './IdeasPage'

describe(`IdeasPage component`, () => {
  it('renders a .IdeasPage by default', () => {
    const wrapper = shallow(<IdeasPage />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders an Idea in an accordion for each ideas in array', () => {
    const props = {
      match: {
        params: {
          listId: 1
        }
      }
    }
    const context = {
      ideas: [
        {
          "id": 1,
          "name": "Idea 1",
          "modified": "2019-01-03T00:00:00.000Z",
          "list_id": 2,
          "content": "Corporis accusamus placeat.\n \rUnde."
        },
        {
          "id": 2,
          "name": "Idea 2",
          "modified": "2018-08-15T23:00:00.000Z",
          "folderId": 1,
          "content": "Eos\n \rlaudantium."
        },
        {
          "id": 3,
          "name": "Idea 3",
          "modified": "2018-03-01T00:00:00.000Z",
          "folderId": 1,
          "content": "Occaecati dignissimos\nvoluptatum nihil."
        },
        {
          "id": 4,
          "name": "Idea 4",
          "modified": "2019-01-04T00:00:00.000Z",
          "folderId": 5,
          "content": "Eum culpa odit."
        },
      ]
    }
    const Accordion = shallow(<IdeasPage {...props} />, context)
      .find('Accordion')
    expect(toJson(Accordion)).toMatchSnapshot()
  })
})