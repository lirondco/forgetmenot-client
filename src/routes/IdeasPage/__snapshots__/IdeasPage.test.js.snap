// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdeasPage component renders a .IdeasPage by default 1`] = `
<Section
  className="IdeasPage"
>
  <div
    className="loading"
  />
</Section>
`;

exports[`IdeasPage component renders an Idea in an accordion for each ideas in array 1`] = `null`;
